main.py

```diff
@@ -18,3 +22,38 @@ def startup():
 @app.get("/health-check")
 def read_root():
     return "ok"
+
+
+@app.get("/inflation")
+def calculate_inflation(type_product: str = "all"):
+    # Selection of products of the same type "food"
+    if type_product == "all":
+        products = collection.find({"type": "food"})
+    else:
+        products = collection.find({"type": "food", "type_product": type_product})
+
+    # Creating a dictionary to store previous prices
+    previous_prices = {}
+
+    # Processing products and linking previous prices to current prices
+    for product in products:
+        product_id = product["id"]  # Product ID
+        current_price = product["price"]  # Current product price
+
+        # Find previous price by product id
+        previous_price = previous_prices.get(product_id)
+        # If the previous price is found, we calculate inflation
+        if previous_price is not None:
+            inflation = ((current_price - previous_price) / previous_price) * 100  # Calculation of inflation
+            product["previous_price"] = previous_price  # Adding "previous_price" field to product
```

Comment:
What is previous_price? This is not correct architecturally. Even despite you don't have `period` of inflation now, this is the feature you'll inevitable add it in a future. You should get prices of previous period and current one and then calculate inflation.

Reply:
Maybe using this [sheet](https://docs.google.com/spreadsheets/d/1UDCPmadOSwIdTeRIjrV9rZKK32Q2GLfR3R_Cpuw5rb4/edit?usp=sharing) it would be easier for you